data-raw/explore_gene_reference.Rmd

```diff
@@ -0,0 +1,120 @@
+sum(symbols_10x2020 %in% scpca_gene_reference$gene_symbol_scpca)
+```
+
+What are they?
+
+```{r}
+# Match the genes that are in the 10x 2020 reference to the ScPCA table on gene symbol
+# Only for those where the gene id is missing in the ScPCA table
+scpca_gene_reference |>
+  filter(
+    !is.na(gene_symbol_10x2020),
+    is.na(gene_symbol_scpca),
+    gene_symbol_10x2020 %in% scpca_gene_reference$gene_symbol_scpca
+  ) |>
+  select(gene_ids, gene_symbol_10x2020, gene_symbol_10x2020_unique) |>
+  left_join(
+    scpca_gene_reference |> select(gene_ids, gene_symbol_scpca, gene_symbol_scpca_unique),
+    by = c("gene_symbol_10x2020" = "gene_symbol_scpca"),
+    suffix = c("_10x2020", "_scpca")
+  )
+```
+
+Looks like these are all cases where the gene id has been updated.
+In only one case is this a gene where something was not unique, which is `LINC01505`.
+This gene (`ENSG00000234229`) seems to have simply been removed in later revisions, so I think we can safely not worry about it.
+
+```{r}
+scpca_gene_reference |>
+  filter(gene_symbol_10x2020 == "LINC01505")
+```
+
+In general, simply translating to the list gene symbols should work as expected.
+
+### BAC removal
+
+In Ensembl version 104, the version we are using for ScPCA, BAC-based gene IDs were removed and replaced with simply the Ensembl gene ID.
+Many of the gene symbols that are present in the 10x references but missing in the ScPCA data are these BAC-based gene IDs.
+We will probably want to translate these to the symbol used in the 10x reference when requested.
+
+```{r}
+scpca_gene_reference |>
+  filter(
+    !is.na(gene_symbol_10x2020),
+    !gene_symbol_10x2020 %in% scpca_gene_reference$gene_symbol_scpca
+  )
+```
+
+### Disagreements between symbols
+
+One other question is how often the `unique` gene symbols disagree for the same Ensemble ID.
+Here we will exclude the cases where the gene symbol is an Ensembl ID in the ScPCA data, as these are expected to be different.
+
+```{r}
+scpca_gene_reference |>
+  filter(
+    gene_symbol_scpca != gene_ids,
+    gene_symbol_scpca_unique != gene_symbol_10x2020_unique
+  )
+```
+
+More than I expected, but it looks like most of these are cases where the gene symbol has been updated.
+We can verify this by looking at how often the base symbol is the same.
+
+```{r}
+scpca_gene_reference |>
+  filter(
+    gene_symbol_scpca != gene_ids,
+    gene_symbol_scpca_unique != gene_symbol_10x2020_unique,
+  ) |>
+  mutate(same_symbol = gene_symbol_scpca == gene_symbol_10x2020) |>
+  count(same_symbol)
+```
+
+What we are really interested in is the cases where the process of making the gene symbol unique had different effects:
+
+```{r}
+scpca_gene_reference |>
+  filter(
+    gene_symbol_scpca == gene_symbol_10x2020,
+    gene_symbol_scpca_unique != gene_symbol_10x2020_unique
+  )
+```
+
+This definitely happens more than I would have hoped, though my hope was probably unreasonable, as 14 cases is not bad.
```

Comment:
14 seems pretty good tbh, or maybe my hopes were unreasonable in the _opposite_ direction 😄